### PR TITLE
Binance: update orderbook, ticker rate limits.

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description
image
I've been getting banned a lot from Binance and these rate limits are why. This PR is non-exhaustive and just looks to fix up the areas where I'm being banned. For more updates, please see #1546 

~I also noticed `GetFundingRateInfo` now just returns `[]` so I removed its usage in the wrapper~

### Please enjoy at your leisure

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Updates test `UpdateTickers` since now it can take in multiple pairs

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance_ufutures.go [hunk 1]
```diff
@@ -94,6 +94,10 @@ func (b *Binance) UFuturesOrderbook(ctx context.Context, symbol currency.Pair, l
 
 	params := url.Values{}
 	params.Set("symbol", symbolValue)
+	if limit == 0 {
+		// default to help select rate limits when no limit is provided
+		limit = 500
+	}
 	strLimit := strconv.FormatInt(limit, 10)
```
Comment (shazbert) [inline exchanges/binance/binance_ufutures.go:101]: This defaults to 500 we don't need to set this if limit is zero

File: exchanges/binance/binance_wrapper.go [hunk 2]
```diff
@@ -578,11 +578,11 @@ func (b *Binance) UpdateOrderbook(ctx context.Context, p currency.Pair, assetTyp
...
 			OrderBookDataRequestParams{
 				Symbol: p,
-				Limit:  1000})
+				Limit:  100})
 	case asset.USDTMarginedFutures:
-		orderbookNew, err = b.UFuturesOrderbook(ctx, p, 1000)
+		orderbookNew, err = b.UFuturesOrderbook(ctx, p, 100)
 	case asset.CoinMarginedFutures:
-		orderbookNew, err = b.GetFuturesOrderbook(ctx, p, 1000)
+		orderbookNew, err = b.GetFuturesOrderbook(ctx, p, 100)
```
Comment (shazbert) [inline exchanges/binance/binance_wrapper.go:585]: Can you please add some explanation on why these changes are needed?

File: exchanges/binance/binance_wrapper.go [hunk 3]
```diff
@@ -2026,22 +2020,14 @@ func (b *Binance) GetLatestFundingRates(ctx context.Context, r *fundingrate.Late
...
-			}
 			nft := time.UnixMilli(mp[i].NextFundingTime)
 			rate := fundingrate.LatestRateResponse{
 				TimeChecked: time.Now(),
 				Exchange:    b.Name,
 				Asset:       r.Asset,
 				Pair:        cp,
 				LatestRate: fundingrate.Rate{
-					Time: time.UnixMilli(mp[i].Time).Truncate(time.Hour * time.Duration(fundingRateFrequency)),
+					Time: time.UnixMilli(mp[i].Time).Truncate(time.Hour),
```
Comment (shazbert) [inline exchanges/binance/binance_wrapper.go:2030]:
```suggestion
					Time: time.UnixMilli(mp[i].NextFundingTime).Add(-time.Hour * 8),
```

As it just retrieves current time. And the ones below?

**Review by shazbert — CHANGES_REQUESTED** [review]
Nice just some things I found while testing.

File: exchanges/binance/binance.go [hunk 4]
```diff
@@ -505,15 +509,43 @@ func (b *Binance) GetPriceChangeStats(ctx context.Context, symbol currency.Pair)
...
+	}
+	if symbolLength > 0 {
+		// this is just easier to format it to  ["x","y"...] than manually
+		symbolsJSON, err := json.Marshal(symbolValues)
+		if err != nil {
+			return resp, err
+		}
+		params.Set("symbols", string(symbolsJSON))
+	}
+	path := priceChange + "?" + params.Encode()
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/binance/binance.go:547]: just easier 😄

File: exchanges/binance/binance_wrapper.go (hunk 2 above)
Comment (gloriousCode) [reply exchanges/binance/binance_wrapper.go:585]: Spot was mostly due to the large budget increase, but there's no need for the others and the spot one is just my own personal one. Will revert 🫡

File: exchanges/binance/binance_ufutures.go (hunk 1 above)
Comment (gloriousCode) [reply exchanges/binance/binance_ufutures.go:101]: Do you mean do this instead?
```go
	if limit == 0 {
		// default
		limit = 500
	}
	strLimit := strconv.FormatInt(limit, 10)
	if !slices.Contains(uValidOBLimits, strLimit) {
		return nil, fmt.Errorf("invalid limit: %v", limit)
	}
	params.Set("limit", strLimit)
```

Comment (gloriousCode) [reply exchanges/binance/binance_ufutures.go:101]: Where does it do this?

File: exchanges/binance/binance.go [hunk 5]
```diff
@@ -505,15 +509,43 @@ func (b *Binance) GetPriceChangeStats(ctx context.Context, symbol currency.Pair)
 	}
 	path := priceChange + "?" + params.Encode()
 
-	return resp, b.SendHTTPRequest(ctx,
-		exchange.RestSpotSupplementary, path, rateLimit, &resp)
+	return resp, b.SendHTTPRequest(ctx, exchange.RestSpotSupplementary, path, rateLimit, &resp)
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/binance/binance.go]: I added those fields, made it a pointer but I think the test change is unnecessary, especially in the context of mock tests

File: exchanges/binance/binance_wrapper.go (hunk 3 above)
Comment (gloriousCode) [reply exchanges/binance/binance_wrapper.go]: Nah, binance has multiple funding rate times. On the upside, the endpoint is back online, so I guess I had a _KNEEJERK REACTION 💥_

File: exchanges/binance/binance.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (shazbert) [reply exchanges/binance/binance.go]: I just wanted to make sure those fields weren't lost in a merge in PR #1546 . But that's fair.

**Review by shazbert — APPROVED** [review]
ch ACK! Nice work.

File: exchanges/binance/binance_wrapper.go [hunk 6]
```diff
@@ -2944,26 +2914,13 @@ func (b *Binance) GetFuturesContractDetails(ctx context.Context, item asset.Item
 		}
 		return resp, nil
 	case asset.CoinMarginedFutures:
-		fri, err := b.GetFundingRateInfo(ctx)
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/binance/binance_wrapper.go:2947]: I will add it back and hope that Binance returns data for it 
# 🙏

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes. Can't believe that just orderbook + ticker updates results in such a gigantic json diff
